pages/docs/core-concepts/create-and-run-effects.mdx

```diff
@@ -106,48 +107,99 @@ Once again inspecting the return type we can see that it is `Effect<never, never
 Sometimes you have to deal with APIs that do not support `async/await` or `Promise`, but are instead implemented in the "old-school" callback style. To support callback-based APIs, Effect provides the `async` constructor.
 
 <Callout type="info">
-Note: Although callback-based APIs may be _considered_ "old-school", they are inherently more powerful than `Promise`-based APIs. This is because they are both faster than `Promise`-based APIs as well as more precise in handling failures.
+  Note: Although callback-based APIs may be _considered_ "old-school", they are
+  inherently more powerful than `Promise`-based APIs. This is because they are
+  both faster than `Promise`-based APIs as well as more precise in handling
+  failures.
 </Callout>
 
 As an example, we can demonstrate wrapping `readFile` from the NodeJS `fs` module with Effect:
 
 ```ts twoslash
-import * as Effect from "@effect/io/Effect"
-import * as NodeFS from "node:fs"
+import * as Effect from "@effect/io/Effect";
+import * as NodeFS from "node:fs";
 
-export const readTodos = Effect.async<never, NodeJS.ErrnoException, Buffer>((resume) => {
-//           ^?
-  NodeFS.readFile("todos.txt", (error, data) => {
-    if (error) {
-      resume(Effect.fail(error))
-    } else {
-      resume(Effect.succeed(data))
-    }
-  })
-})
+export const readTodos = Effect.async<never, NodeJS.ErrnoException, Buffer>(
+  //              ^?
+  (resume) => {
+    NodeFS.readFile("todos.txt", (error, data) => {
+      if (error) {
+        resume(Effect.fail(error));
+      } else {
+        resume(Effect.succeed(data));
+      }
+    });
+  }
+);
 ```
 
-Something to note in the above example is that we were actually forced to manually annotate the types when calling `Effect.async`. Unfortunately, TypeScript is not capable of inferring the type parameters for a callback based on what is returned inside the body of the callback. This limitation of TypeScript is the one minor drawback of using callback-based APIs.
+Something to note in the above example is that we were actually forced to manually annotate the types when calling `Effect.async`. Unfortunately, TypeScript is not capable of inferring the type parameters for a callback based on what is returned inside the body of the callback. This limitation of TypeScript is the one minor drawback of using callback-based APIs. However, it will validate that the types you passed in match what is being returned.
```

Comment:
I would maybe change this to say: "Annotating the types of `Effect.async` will ensure that the values provided to `resume` match the expected types.".

Reply:
Thanks!